react/__tests__/Slider.test.tsx

```diff
@@ -0,0 +1,319 @@
+import React from 'react'
+import { render } from '@vtex/test-tools/react'
+
+import Slider from '../components/Slider'
+import { mockInitialInfiniteSliderState } from '../__fixtures__/SliderStateContext'
+
+let mockDeviceDetectorReturn = { device: 'desktop', isMobile: false }
+
+jest.mock('../hooks/useTouchHandlers', () => ({
+  useTouchHandlers: () => ({
+    onTouchStart: jest.fn(),
+    onTouchMove: jest.fn(),
+    onTouchEnd: jest.fn(),
+  }),
+}))
+
+jest.mock('../components/SliderContext', () => ({
+  useSliderState: () => mockInitialInfiniteSliderState,
+  useSliderDispatch: () => jest.fn(),
+}))
+
+jest.mock('vtex.device-detector', () => ({
+  useDevice: jest.fn(() => mockDeviceDetectorReturn),
+}))
+
+describe('Basic rendering', () => {
+  it('should render complete slider, with arrows and pagination dots on default settings', () => {
+    const { getByTestId, getByLabelText } = render(
+      <Slider
+        totalItems={10}
+        itemsPerPage={5}
+        showNavigationArrows="always"
+        showPaginationDots="always"
+        arrowSize={25}
+        fullWidth
+        infinite
+        usePagination
+      />
+    )
+
+    const expectedCarouselRole = getByLabelText('slider')
+    const leftArrow = getByTestId('icon-caret-left')
+    const rightArrow = getByTestId('icon-caret-right')
+    const paginationDots = getByLabelText('Slider pagination dots')
+
+    expect(expectedCarouselRole).toBeDefined()
+    expect(leftArrow).toBeDefined()
+    expect(rightArrow).toBeDefined()
+    expect(paginationDots).toBeDefined()
+  })
+
+  it('should render without pagination features and use x-scrolling if usePagination is set to false', () => {
+    const { getByLabelText, queryByTestId, queryByLabelText } = render(
+      <Slider
+        totalItems={10}
+        itemsPerPage={5}
+        showNavigationArrows="always"
+        showPaginationDots="always"
+        usePagination={false}
+        arrowSize={25}
+        fullWidth
+        infinite
+      />
+    )
+
+    const renderedSlider = getByLabelText('slider')
+    const sliderTrackContainer = renderedSlider.firstChild
+
+    const leftArrow = queryByTestId('icon-caret-left')
+    const rightArrow = queryByTestId('icon-caret-right')
+    const paginationDots = queryByLabelText('Slider pagination dots')
+
+    expect(leftArrow).toBeNull()
+    expect(rightArrow).toBeNull()
+    expect(paginationDots).toBeNull()
+    expect(
+      (sliderTrackContainer as HTMLElement | null)?.className.split(' ')
+    ).toContain('overflow-x-scroll')
```

Comment:
Update to the lastest version of test tools you might use the matcher toHaveClass

```suggestion
    expect(sliderTrackContainer).toHaveClass('overflow-x-scroll')
```

Other matchers
https://github.com/testing-library/jest-dom

Reply:
Awesome!